### PR TITLE
[INLONG-2250][Feature][InLong-Sort] Support Kafka sink in InLong-Sort 

### LINKED ISSUE #2250: [Feature][InLong-Sort] Support Kafka sink in InLong-Sort
### Description

Support Kafka sink in InLong-Sort.

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by KevinWen007**
### Title Name: [INLONG-2250][Feature][InLong-Sort] Support Kafka sink in InLong-Sort

Fixes #2250 

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve.*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

File: inlong-sort/sort-core/pom.xml [hunk 1]
```diff
@@ -227,6 +227,11 @@
             <version>3.1</version>
         </dependency>
 
+        <dependency>
+            <groupId>org.apache.flink</groupId>
+            <artifactId>flink-connector-kafka_2.11</artifactId>
```
Comment (chantccc) [inline inlong-sort/sort-core/pom.xml:232]: Currently, sort-core do not need to depend on this

File: inlong-sort/sort-single-tenant/src/main/java/org/apache/inlong/sort/singletenant/flink/Entrance.java [hunk 2]
```diff
@@ -93,7 +100,9 @@ private static DataFlowInfo getDataflowInfoFromFile(String fileName) throws IOEx
     private static void buildSinkStream(
             DataStream<Row> sourceStream,
             Configuration config,
-            SinkInfo sinkInfo) {
+            SinkInfo sinkInfo,
+            Map<String, Object> properties,
+            CheckpointingMode checkpointingMode) {
```
Comment (chantccc) [inline inlong-sort/sort-single-tenant/src/main/java/org/apache/inlong/sort/singletenant/flink/Entrance.java:105]: put checkpointingMode in config, and give it a default value 'EXACTLY_ONCE'

File: inlong-sort/sort-core/src/main/java/org/apache/inlong/sort/flink/kafka/KafkaSerializationSchemaBuilder.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+package org.apache.inlong.sort.flink.kafka;
+
+import org.apache.flink.api.common.serialization.SerializationSchema;
+import org.apache.flink.types.Row;
+import org.apache.inlong.sort.protocol.serialization.SerializationInfo;
+
+import java.nio.charset.StandardCharsets;
+
+public class KafkaSerializationSchemaBuilder {
```
Comment (chantccc) [inline inlong-sort/sort-core/src/main/java/org/apache/inlong/sort/flink/kafka/KafkaSerializationSchemaBuilder.java:26]: KafkaSerializationSchemaBuilder -> SerializationSchemaBuilder
the builder will be used in more sinks in the future

**Review by chantccc — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Comment by healchow** [comment]
Is it possible to write unit tests to avoid this kind of error?

**Review by healchow — APPROVED** [review]

**Pull request merged**